bindings/CMakeLists.txt

```diff
@@ -50,29 +50,21 @@ endif("${SWIG_VERSION}" VERSION_LESS "${MIN_SWIG_VERSION}")
 set_source_files_properties(icub.i PROPERTIES CPLUSPLUS ON)
 
 if(CREATE_PYTHON)
+    find_package(Python REQUIRED COMPONENTS Interpreter Development)
     set(CMAKE_SWIG_FLAGS "-Wall;-module;icub;-threads")
 
-    set(ICUB_USE_PYTHON_VERSION "" CACHE STRING "Specify python version to use" )
-
-    find_package(PythonInterp ${ICUB_USE_PYTHON_VERSION} REQUIRED)
-    set(ICUB_USE_PYTHON_VERSION_ARGS)
-    if(NOT ICUB_USE_PYTHON_VERSION)
-      set (ICUB_USE_PYTHON_VERSION ${PYTHON_VERSION_STRING})
-    endif()
-    find_package(PythonLibs ${ICUB_USE_PYTHON_VERSION} EXACT)
-
-    include_directories(SYSTEM ${PYTHON_INCLUDE_DIRS})
+    include_directories(SYSTEM ${Python_INCLUDE_DIRS})
     swig_add_library(icub_python
                      LANGUAGE python
                      SOURCES icub.i)
-    target_include_directories(${SWIG_MODULE_icub_python_REAL_NAME} SYSTEM PRIVATE ${PYTHON_INCLUDE_DIRS})
-    target_link_libraries(${SWIG_MODULE_icub_python_REAL_NAME} ${PYTHON_LIBRARIES})
+    target_include_directories(${SWIG_MODULE_icub_python_REAL_NAME} SYSTEM PRIVATE ${Python_INCLUDE_DIRS})
+    target_link_libraries(${SWIG_MODULE_icub_python_REAL_NAME} ${Python_LIBRARIES})
     set_target_properties(${SWIG_MODULE_icub_python_REAL_NAME} PROPERTIES OUTPUT_NAME "_icub")
 
     # installation path is determined reliably on most platforms using distutils
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix=''))"
+    execute_process(COMMAND ${Python_EXECUTABLE} -c "from distutils import sysconfig; print(sysconfig.get_python_lib(1,0,prefix=''))"
                     OUTPUT_VARIABLE PYTHON_INSTDIR
-                    OUTPUT_STRIP_TRAILING_WHITESPACE )
+                    OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
For the time being this is ok, however let's keep in mind that the new FindPython provides something similar. Though, there are few caveats as discussed in https://github.com/robotology/robotology-superbuild/pull/359#issuecomment-592956444.

Reply:
Thanks for pointing this out, although I would rather stay more conservative here 👍 
Feel free to contribute in this sense anyway.